src/ui/public/vislib/lib/axis/axis_config.js

```diff
@@ -156,7 +156,9 @@ export default function AxisConfigFactory() {
 
     get(property, defaults) {
```

Comment:
this method is getting seriously wicked with all the exit points :)

I'd consider rejiggering a little.

e.g.:

```
if (typeof defaults === 'undefined' && !_.has(this._values, property)) {
 throw new Error(`Accessing invalid config property: ${property}`);
}

const val = _.get(this._values, property, defaults);
return (val == null && defaults != null) ? defaults : val;
```